prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5245,6 +5360,40 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         Ok(stmts)
     }
 
+    fn encode_slice_types(
+        &self,
+        slice_ty: ty::Ty<'tcx>,
+    ) -> EncodingResult<(
+        String,
+        vir::Type,
+        String,
+        vir::Type,
+        vir::Type,
+        ty::Ty<'tcx>,
+    )> {
+        // get rid of ref
+        let slice_ty = match slice_ty.kind() {
+            ty::TyKind::Ref(_, ty, _) => ty,
```

Comment:
If it's known at the call-site of `encode_slice_types` that `slice_ty` is a `TyKind::Ref`, then I would prefer to remove this match and move it to the caller.

Reply:
ack, moved.